patches/sf17-79.patch

```diff
@@ -107,6 +107,19 @@ index 15208255..7f9f7dfa 100644
      bool read_header(std::istream&, std::uint32_t*, std::string*) const;
      bool write_header(std::ostream&, std::uint32_t, const std::string&) const;
  
+diff --git a/src/timeman.h b/src/timeman.h
+index 10207a8a..403006a5 100644
+--- a/src/timeman.h
++++ b/src/timeman.h
+@@ -48,7 +48,7 @@ class TimeManagement {
+     TimePoint elapsed(FUNC nodes) const {
+         return useNodesTime ? TimePoint(nodes()) : elapsed_time();
+     }
+-    TimePoint elapsed_time() const { return now() - startTime; };
++    TimePoint elapsed_time() const { return std::max(now() - startTime, TimePoint(1)); };
```

Comment:
im curious about this change, why did you need that ? We are using `std::chrono::steady_clock` and it is supposed to be monotonic ? 
> Class std::chrono::steady_clock represents a monotonic clock. The time points of this clock cannot decrease as physical time moves forward and the time between ticks of this clock is constant.

Reply:
Unfortunately emscripten is failing to reliably provide monotonicity across threads in separate web workers.

* After https://github.com/emscripten-core/emscripten/issues/9783 it's roughly correct, at least
* https://github.com/emscripten-core/emscripten/issues/15380 is still open, though

Reply:
Changed to `TimePoint(0)`, which is more precise (5ccdf71971099190a351910d7a9df6c59a8f73db)

Reply:
Oh yikes, thanks...

Reply:
we do use PROXY_TO_PTHREAD, is it still an issue?

Reply:
Unless there's been further patches, each worker asks the parent thread for an offset/skew when it's first created. That's rarely off by more than a few milliseconds, but still doesn't guarantee monotonicity in all cases (measure in parent after child, measure in child a after child b).

Proxying every single time call to a specific thread would work, but I don't think that was ever implemented.